packages/cms-data-sync/src/news/news.data-migration.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable no-console */
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+import { Entry, SysLink } from 'contentful-management';
+import { Document } from '@contentful/rich-text-types';
+
+import { newsQuery } from './news.queries';
+import {
+  Asset,
+  FetchNewsQuery,
+  FetchNewsQueryVariables,
+  Maybe,
+} from '../autogenerated-gql/graphql';
+
+import { migrateAsset } from '../assets/assets.data-migration';
+import {
+  convertText,
+  clearContentfulEntries,
+  getSquidexAndContentfulClients,
+  publishContentfulEntries,
+} from '../utils';
+
+type NewsPayload = {
+  title: { 'en-US': string };
+  shortText: { 'en-US': string | null };
+  frequency: { 'en-US': string | null };
+  link: { 'en-US': string | undefined };
+  linkText: { 'en-US': string | undefined };
+  text: { 'en-US': Document | null };
+  thumbnail: { 'en-US': SysLink | null };
+};
+
+const migrateNews = async () => {
+  const { contentfulEnvironment, squidexGraphqlClient } =
+    await getSquidexAndContentfulClients();
+
+  const convertThumbnail = async (
+    newsPayload: NewsPayload,
+    thumbnail: Maybe<
+      Pick<
+        Asset,
+        'id' | 'fileName' | 'thumbnailUrl' | 'mimeType' | 'fileType'
+      >[]
+    >,
+  ) => {
+    if (thumbnail?.length) {
+      await migrateAsset(contentfulEnvironment, thumbnail);
+      newsPayload.thumbnail = {
+        'en-US': {
+          sys: {
+            type: 'Link',
+            linkType: 'Asset',
+            // in thumbnail it's possible to load only one asset
+            id: thumbnail[0]?.id,
+          },
+        },
+      };
+    }
+  };
+
+  const gqlNews = await squidexGraphqlClient.request<
```

Comment:
Replying in here so we have the answer in github as well :)

I haven't use the data provider because in rest the thumbnail didn't come with its filename, mimeType, etc and we use this to upload the asset in contentful